### PR TITLE
feat: add security header

### PR DESCRIPTION
**Pull request by zaunist**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/ #2340 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by zaunist** [comment]
@nic-chen @bzp2010 PTAL, thanks.

**Review by nic-chen — APPROVED** [review]

File: api/conf/conf.yaml [hunk 1]
```diff
@@ -60,6 +60,14 @@ conf:
                          # such as absolute path on Windows: winfile:///C:\access.log
                          # log example: 2020-12-09T16:38:09.039+0800	INFO	filter/logging.go:46	/apisix/admin/routes/r1	{"status": 401, "host": "127.0.0.1:9000", "query": "asdfsafd=adf&a=a", "requestId": "3d50ecb8-758c-46d1-af5b-cd9d1c820156", "latency": 0, "remoteIP": "127.0.0.1", "method": "PUT", "errs": []}
   max_cpu: 0             # supports tweaking with the number of OS threads are going to be used for parallelism. Default value: 0 [will use max number of available cpu cores considering hyperthreading (if any)]. If the value is negative, is will not touch the existing parallelism profile.
+  # security:
+  #   access_control_allow_origin: "http:httpbin.org"
```
Comment (starsz) [inline api/conf/conf.yaml:64]:
```suggestion
  #   access_control_allow_origin: "http://httpbin.org"
```

File: api/conf/conf.yaml [hunk 2]
```diff
@@ -60,6 +60,14 @@ conf:
...
+  #   access_control_allow_credentials: true          # support user custom cors configration
```
Comment (starsz) [inline api/conf/conf.yaml:65]:
```suggestion
  #   access_control_allow_credentials: true          # support using custom cors configration
```

File: api/internal/filter/cors.go [hunk 3]
```diff
@@ -16,14 +16,36 @@
  */
 package filter
 
-import "github.com/gin-gonic/gin"
+import (
+	"github.com/apisix/manager-api/internal/conf"
```
Comment (starsz) [inline api/internal/filter/cors.go:20]: Code style.

File: api/internal/conf/conf.go [hunk 4]
```diff
@@ -316,3 +330,24 @@ func initParallelism(choiceCores int) {
...
+		if conf.XFrameOptions == "" {
+			SecurityConf.XFrameOptions = "deny"
+		}
+		return
+	}
+
+	SecurityConf = Security{
+		XFrameOptions:         "deny",
+		ContentSecurityPolicy: "default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'",
+	}
```
Comment (starsz) [inline api/internal/conf/conf.go:352]: So we don't have the default value of `AllowCredentials` `AllowOrigin` `AllowMethods` `AllowHeaders`

Comment (zaunist) [reply api/internal/conf/conf.go:352]: We don't need default value for `AllowCredentials`、`AlloOrigin`、`AllowMethods`、`AllowHeaders`. These configuration are left to the user to customize.

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**